### PR TITLE
More tests to improve coverage and fix LiveTests

### PR DESCRIPTION
**Pull request by arifsaikat-microsoft**

### CODE REVIEW

File: sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallConnectionAsync.java [hunk 1]
```diff
@@ -162,30 +162,6 @@ public String getCallConnectionId() {
         return playAudioWithResponseInternal(audioFileUri, playAudioOptions, null);
     }
 
-    Mono<Response<PlayAudioResult>> playAudioWithResponseInternal(
```
Comment (minnieliu) [inline sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallConnectionAsync.java:165]: Why was this removed?

File: sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallingServerTestBase.java [hunk 2]
```diff
@@ -109,7 +112,17 @@ protected String getGroupId(String testName) {
           have unique groupId's so they do not conflict with other
           recording tests running in live mode.
          */
-        if (getTestMode() == TestMode.LIVE) {
+        if (getTestMode() == TestMode.LIVE) {        
+            return UUID.randomUUID().toString();
+        }
+
+        /*
+          The Live test pipeline runs in recording mode, so we
```
Comment (minnieliu) [inline sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallingServerTestBase.java:120]: This isn't true, the live test pipeline runs in LIVE mode. What is the expected behaviour that you are trying to accomplish?

File: sdk/communication/azure-communication-callingserver/tests.yml [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+        - task: PublishCodeCoverageResults@1
+          condition: and(eq(variables['Agent.OS'], 'Windows_NT'), eq(variables['JavaTestVersion'], '1.11'), startsWith(variables['System.StageName'], 'Public'))
+          inputs:
+            codeCoverageTool: JaCoCo
+            summaryFileLocation: sdk/communication/azure-communication-callingserver/target/site/jacoco/jacoco.xml
+            reportDirectory: sdk/communication/azure-communication-callingserver/target/site/jacoco/
+            failIfCoverageEmpty: true
+      EnvVars:
+        SKIP_LIVE_TEST: TRUE
+        CALLINGSERVER_GENERATE_RANDOM_GROUP_IDENTIFIER: TRUE
```
Comment (benbp) [inline sdk/communication/azure-communication-callingserver/tests.yml:31]: Are you copying the whole template so you can add the extra env var and/or remove int from the cloud config? I think it would be better to add support for passing these both in as parameters, e.g.:

tests.yml
```
 - template: /sdk/communication/communication-tests-template.yml
     parameters:
       Clouds: Public
       EnvVars:
         CALLINGSERVER_GENERATE_RANDOM_GROUP_IDENTIFIER: TRUE
```

/sdk/communication/communication-tests-template.yml
```
parameters:
  PackageName: not-specified
  SafeName: notspecified
  Cloud: 'Public,Int'
  EnvVars: {}

stages:
  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
    parameters:
      Clouds: ${{ parameters.Clouds }}

... other stuff ...

      EnvVars:
        SKIP_LIVE_TEST: TRUE
        ${{ each var in parameters.EnVars }}:
          ${{ var.key }}: ${{ var.value }}

```

File: sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallConnectionAsync.java (hunk 1 above)
Comment (arifsaikat-microsoft) [reply sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/CallConnectionAsync.java:165]: This was not being used, so removed. 

To give a bit background, we used to have 4 version of playAudio() service methods:
PlayAudioResult playAudio(String audioFileUri, boolean loop, String audioFileId, String callbackUri, String perationContext);
Response<PlayAudioResult> playAudio(String audioFileUri, boolean loop, String audioFileId, String callbackUri, String perationContext, final Context context);
PlayAudioResult playAudio(String audioFileUri, PlayAudioOptions playAudioOptions);
Response<PlayAudioResult> playAudioWithResponse(String audioFileUri, PlayAudioOptions playAudioOptions,
        final Context context);
        
Based on one of the feedbacks on ApiView from Srikanta that we need only one method with Response, we removed the second one, thus this internal one became unused.

File: sdk/communication/azure-communication-callingserver/tests.yml (hunk 3 above)
Comment (chrwhit) [reply sdk/communication/azure-communication-callingserver/tests.yml:31]: Let me try that out

File: sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallingServerTestBase.java (hunk 2 above)
Comment (chrwhit) [reply sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallingServerTestBase.java:120]: Ours seems to run in Record, maybe it is configured incorrectly

Comment (chrwhit) [reply sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallingServerTestBase.java:120]: [Link to Live Pipeline test run](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=960406&view=logs&j=186fb226-6090-5bc4-037b-dfee8c37ef57&t=24176779-b866-57d9-80d3-8324e431459f&l=19)

Comment (minnieliu) [reply sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallingServerTestBase.java:120]: Sorry I just double checked, you are right, it is in record mode.

Comment (chrwhit) [reply sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallingServerTestBase.java:120]: I just talked with Alan Zimmer, and it seems the tests were temporarily switched to RECORD because LIVE was causing issues in azure-core-tests. This is going to be changed back soon. I will look at removing this work around for our tests, and also forcing LIVE mode for these tests, just incase there is another change.

**Comment by minnieliu** [comment]
What is the test coverage on calling after the new tests have been added now? 
It would be good to set min code coverage if you haven't already: 
image

File: sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallingServerTestBase.java (hunk 2 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/CallingServerTestBase.java:120]: Sounds good, thanks for clarifying.

**Comment by arifsaikat-microsoft** [comment]
> What is the test coverage on calling after the new tests have been added now?
> It would be good to set min code coverage if you haven't already:
> image

After new tests added, current line coverage is 82% and branch coverage is 84%.

We already have set min code coverage to 70%
https://github.com/Azure/azure-sdk-for-java/blob/58e6e58a229e05ff60e3aa606cc28e30d6b3c21a/sdk/communication/azure-communication-callingserver/pom.xml#L43

File: sdk/communication/azure-communication-callingserver/tests.yml (hunk 3 above)
Comment (chrwhit) [reply sdk/communication/azure-communication-callingserver/tests.yml]: Fixed.

File: sdk/communication/azure-communication-callingserver/tests.yml [hunk 4]
```diff
@@ -1,8 +1,32 @@
 trigger: none
 
 stages:
-  - template: /sdk/communication/communication-tests-template.yml
+  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
     parameters:
-      PackageName: azure-communication-callingserver
-      SafeName: azurecommunicationcallingserver
```
[the remark this answers is not in the document]
Comment (chrwhit) [reply sdk/communication/azure-communication-callingserver/tests.yml:7]: Benbp left similar feedback, we were able to do this, but needed to pipe it through communication-tests-template.yml.

**Review by JianpingChen — APPROVED** [review]

**Review by benbp — APPROVED** [review]

**Review by minnieliu — APPROVED** [review]
